src/App.js

```diff
@@ -10,8 +10,9 @@ import Details from './Pages/Details';
 
 function App() {
   const [characters, setCharacters] = useState([]);
-  const [favoriteIDs, setFavoriteIDs] = useState(JSON.parse(localStorage.getItem('favoriteIDs')) || []
-  )
+  const [favoriteIDs, setFavoriteIDs] = useState(
+    JSON.parse(localStorage.getItem('favoriteIDs')) || []
```

Comment:
It's better to put this into a useEffect hook that is only being executed once on initial render. You should not put expensive calculations or something similar into the useState hook as initial state, see docs: " If the initial state is the result of an expensive computation, you may provide a function instead, which will be executed only on the initial render"
The docs say you can basically put this in as inital state, but you have to wrap it into a function, which is an alternative to useEffect int hat case.